Explorer/Assets/DCL/Infrastructure/ECS/Unity/GltfNodeModifiers/Systems/UpdateGltfNodeModifierSystem.cs

```diff
@@ -0,0 +1,151 @@
+        }
+
+        /// <summary>
+        ///     Updates the global modifier (single modifier with empty path)
+        /// </summary>
+        private void UpdateGlobalModifier(Entity containerEntity, PBGltfNodeModifiers.Types.GltfNodeModifier modifier, ref GltfContainerComponent gltfContainer, GltfContainerAsset asset)
+        {
+            // Check if transitioning from individual modifiers to global modifier
+            if (gltfContainer.GltfNodeEntities is { Count: > 0 } &&
+                (gltfContainer.GltfNodeEntities.Count > 1 || gltfContainer.GltfNodeEntities[0] != containerEntity))
+            {
+                // Clean up individual modifier entities
+                foreach (Entity entityToCleanup in gltfContainer.GltfNodeEntities) { CleanupGltfNodeEntity(entityToCleanup, containerEntity); }
+
+                gltfContainer.GltfNodeEntities.Clear();
+
+                // Add global GltfNode component to container entity if transitioning from individual
+                if (!World.Has<GltfNode>(containerEntity)) { CreateGlobalGltfNode(containerEntity, asset); }
+
+                gltfContainer.GltfNodeEntities!.Add(containerEntity);
+            }
+
+            (bool hasShadowOverride, bool hasMaterialOverride) = GetModifierOverrides(modifier);
+
+            asset.SetCastingShadows(!hasShadowOverride || modifier.CastShadows);
+
+            if (hasMaterialOverride) { AddOrUpdateMaterial(containerEntity, modifier.Material); }
+            else
+            {
+                if (World.Has<PBMaterial>(containerEntity))
+                {
+                    // Material was removed, clean it up but don't destroy entity
+                    GltfNode gltfNode = World.Get<GltfNode>(containerEntity);
+                    CreateMaterialCleanupIntention(containerEntity, gltfNode.Renderers, gltfNode.ContainerEntity, false);
+                }
+            }
+        }
+
+        /// <summary>
+        ///     Updates individual modifiers (multiple modifiers or specific paths)
+        /// </summary>
+        private void UpdateIndividualModifiers(Entity containerEntity, IList<PBGltfNodeModifiers.Types.GltfNodeModifier> modifiers, ref GltfContainerComponent gltfContainer, PartitionComponent partitionComponent, GltfContainerAsset asset)
+        {
+            if (gltfContainer.GltfNodeEntities == null) return;
+
+            // Check if transitioning from global modifier to individual modifiers
+            if (gltfContainer.GltfNodeEntities.Count == 1 && gltfContainer.GltfNodeEntities[0] == containerEntity)
+            {
+                CleanupGltfNodeEntity(containerEntity, containerEntity);
+                gltfContainer.GltfNodeEntities.Remove(containerEntity);
+            }
+
+            var existingGltfNodePaths = new Dictionary<string, Entity>();
+
+            foreach (Entity nodeEntity in gltfContainer.GltfNodeEntities) { existingGltfNodePaths[World.Get<GltfNode>(nodeEntity).Path!] = nodeEntity; }
```

Comment:
It can produce hiccups as `World.Get<GltfNode>(nodeEntity).Path` is `O(N^2)` (for every Modifies definition for every contained entity).

You should be thinking about modifying the algorithm or introducing budgeting at least on the level of `UpdateGltfNodes`